MHVLibTests/Connection/MHVConnectionTests.m

```diff
@@ -412,16 +398,174 @@ @interface MHVConnection (Testing)
                 it(@"url should be formatted with path from pathParams and queryParams", ^
                    {
                        [[expectFutureValue(tokenWasRefreshed) shouldEventually] beYes];
-                       [[expectFutureValue(requestedURL.absoluteString) shouldEventually] equal:@"https://test.url/path?query1=ABC"];
+                       [[expectFutureValue(requestedURL.absoluteString) shouldEventually] equal:@"https://rest.url/path?query1=ABC"];
                    });
                 
                 it(@"refreshed authorization header should be set", ^
                    {
                        [[expectFutureValue(tokenWasRefreshed) shouldEventually] beYes];
                        [[expectFutureValue(requestedHeaders) shouldEventually] beNonNil];
-                       [[expectFutureValue(requestedHeaders[@"Authorization"]) shouldEventually] equal:@"REFRESHED-TOKEN"];
+                       [[expectFutureValue(requestedHeaders[@"Authorization"]) shouldEventually] equal:[NSString stringWithFormat:kAuthorizationHeaderFormat, kRefreshedToken]];
                    });
             });
+    
+    context(@"MHVMethod reissued for 500 errors", ^
+            {
+                beforeEach(^{
+                    requestCompletionError = [NSError MHVUnknownError];
+                    requestCompletionResponse = [[MHVHttpServiceResponse alloc] initWithResponseData:nil statusCode:500];
+                    
+                    MHVMethod *method = [MHVMethod getThings];
+                    method.parameters = @"GETTHINGSBODY";
+                    [testConnection executeHttpServiceOperation:method
+                                                     completion:^(MHVServiceResponse * _Nullable response, NSError * _Nullable error) { }];
+                });
+                
+                it(@"requestCount should be 3", ^
+                   {
+                       [[expectFutureValue(requestCount) shouldEventuallyBeforeTimingOutAfter(10)] equal:@(3)];
```

Comment:
I think we also need to make sure that requests don't end up in an infinite loop. This test will succeed once the requestCount variable == 3, but if we ned to make sure it never reaches 4.